### PR TITLE
FED-734 CI cleanup: remove failing stable config, add 2.18 config

### PR DESCRIPTION
**Pull request by greglittlefield-wf**
## Motivation
- The GitHub Actions "build" job, which is our main set of CI checks, hasn't only been running on 2.13.4 for a long time, and nothing newer. I'd really like to include a build for 2.18
- The "validate_analyzer" job is failing on Dart stable due to a Dart SDK bug that we're unable to work around at this time, so we want to disable it.

## Changes
- Add 2.18.7 to the build matrix, since it's the latest Dart SDK version we can build with
   - Fix test failures in newer SDKs
   - Ignore `part_of_different_library` errors
   - Upgrade mockito since it older versions were causing build failures (merge in https://github.com/Workiva/over_react/pull/789) - thanks @evanweible-wf!
   - Fix usages of the `dartanalyzer` command removed in newer SDKs
   - Fix VM test command to not use `build_runner`
   - Update built_redux/mockito generated file checks to only run in 2.18.7
- Remove validate_analyzer config, since it isn't needed once #804 constrains our analyzer dependency to a single major version
    - We can reinstate this if we ever need to support multiple versions again in the future

## Review
_[See CONTRIBUTING.md][contributing-review-types] for more details on review types (+1 / QA +1 / +10) and code review process._

  

Please review: 

### QA Checklist
- [ ] Tests were updated and provide good coverage of the changeset and other affected code
- [ ] Manual testing was performed if needed
    - [ ] Steps from PR author: 
        - CI passes
    - [ ] Anything falling under manual testing criteria [outlined in CONTRIBUTING.md][contributing-manual-testing]

## Merge Checklist
While we perform many automated checks before auto-merging, some manual checks are needed:
- [ ] A Frontend Frameworks Design member has reviewed these changes
- [ ] There are no unaddressed comments _- this check can be automated if reviewers use the "Request Changes" feature_
- [ ] _For release PRs -_ Version metadata in Rosie comment is correct

[contributing-review-types]: https://github.com/Workiva/over_react/blob/master/CONTRIBUTING.md#review-types
[contributing-manual-testing]: https://github.com/Workiva/over_react/blob/master/CONTRIBUTING.md#manual-testing-criteria

### CODE REVIEW

**Comment by aviary-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: app/over_react_redux/todo_client/pubspec.yaml [hunk 1]
```diff
@@ -7,7 +7,7 @@ environment:
 dependencies:
   color: any
   memoize: ^2.0.0
-  meta: ^1.6.0 # Workaround to avoid https://github.com/dart-lang/sdk/issues/46142
```
Comment (greglittlefield-wf) [inline app/over_react_redux/todo_client/pubspec.yaml:10]: These comments were outdated, so I cleaned them up

File: test/over_react/util/react_wrappers_test.dart [hunk 2]
```diff
@@ -1063,19 +1063,25 @@ main() {
               expect(result2, same(result1), reason: 'should have returned the same object');
             }, tags: 'no-ddc');
 
-            test('unless the runtime is the DDC', () {
```
Comment (greglittlefield-wf) [inline test/over_react/util/react_wrappers_test.dart:1066]: Basically, before we were testing that caching wasn't happening in DDC, which wasn't really the behavior we wanted ensure. I replaced it with a test that validates that it doesn't break, which is really the behavior we're after, at least for older SDKs.

**Review by robbecker-wf — APPROVED** [review]
Looks good. Thanks for the detailed comments on why things were done.

**Review by evanweible-wf — APPROVED** [review]

**Comment by robbecker-wf** [comment]
QA+1 CI Passes on 2.13 and 2.18
tests updated and pass with newer mockito

**Comment by greglittlefield-wf** [comment]
@Workiva/release-management-p

**Pull request merged**